packages/std/src/results/response.rs

```diff
@@ -54,21 +49,19 @@ use super::{Attribute, CosmosMsg, Empty, Event, SubMsg};
 ///     info: MessageInfo,
 ///     msg: InstantiateMsg,
 /// ) -> Result<Response, MyError> {
-///     let mut response = Response::new();
-///     // ...
-///     response.add_attribute("Let the", "hacking begin");
-///     // ...
-///     response.add_message(BankMsg::Send {
-///         to_address: String::from("recipient"),
-///         amount: coins(128, "uint"),
-///     });
-///     response.add_attribute("foo", "bar");
-///     // ...
-///     response.set_data(Binary::from(b"the result data"));
+///     let mut response = Response::new()
+///         .add_attribute(("Let the", "hacking begin"))
+///         .add_message(BankMsg::Send {
+///             to_address: String::from("recipient"),
+///             amount: coins(128, "uint"),
+///         })
+///         .add_attribute(("foo", "bar"))
+///         .set_data(b"the result data");
 ///     Ok(response)
 /// }
 /// ```
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[non_exhaustive]
```

Comment:
Didn't knew that `non_exhaustive` was there for structs too. Nice.

Reply:
Are we still able to use assignments for clearing things? E.g. `response.messages = vec![]`?

Reply:
Yes! The user can mutate the struct's fields. The only things they can't do are
* construct it via a struct literal,
* do a destructuring assignment without `..`

Reply:
Very cool trick. accomplishing basically what the ZST would do, with a compiler flag